src/components/DetailedArticle.jsx

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getArticleById } from "../api";
+import ArticleCard from "./ArticleCard";
+
+function DetailedArticle() {
+  const { article_id } = useParams();
+
+  const [article, setArticle] = useState(null);
+
+  useEffect(() => {
+    getArticleById(article_id).then((response) => {
+      setArticle(response.data.article);
+    });
+  }, [article_id]);
+
+  if (!article) {
+    return <p>Loading...</p>;
+  }
```

Comment:
It's really good that you've got a functional loading state to conditionally render while the fetch is still ongoing, however, I'd say it's a bad practice to have your `article` state change from `null` ( a primitive data type ) to an `object`, check out [these notes](https://notes.northcoders.com/courses/js-front-end/useEffect#loading-patterns) for using state to handle _loading_